src/common/Gateway.sol

```diff
@@ -275,8 +273,15 @@ contract Gateway is Auth, Recoverable, IGateway {
             }
         }
 
-        batcher = address(0);
         _endBatching(refund);
+        _batcher = address(0);
+    }
+
+    /// @inheritdoc IGateway
+    function lockCallback() external returns (address caller) {
```

Comment:
Is it even helpful to return the `caller` here? Should always be the callback destination right?